ops/ccs-ops-misc/load_test/common/stats.py

```diff
@@ -0,0 +1,274 @@
+
+        Returns:
+            Dict[str, int]: A dictionary of human-readable percentile strings to the percent response time
+        """
+        if not stats_entry.num_requests:
+            return self.percentiles_na
+
+        return {self._get_readable_percentile(percentile): int(stats_entry.get_response_time_percentile(percentile) or 0) for percentile in self.percentiles_to_report}
+
+    def _get_stats_entry_dict(self, stats_entry: StatsEntry) -> Dict[str, any]:
+        """Returns a dictionary representation of a StatsEntry object
+
+        Args:
+            stats_entry (StatsEntry): The Locust StatsEntry object which encodes a particular task's statistics
+
+        Returns:
+            Dict[str, any]: A dictionary that represents most of the statistics exposed by StatsEntry, including percentiles
+        """
+        return {**{
+            'name': stats_entry.name,
+            'requestMethod': stats_entry.method,
+            'numRequests': stats_entry.num_requests,
+            'numFailures': stats_entry.num_failures,
+            'medianResponseTime': stats_entry.median_response_time,
+            'avgResponseTime': stats_entry.avg_response_time,
+            'minResponseTime': stats_entry.min_response_time or 0,
+            'maxResponseTime': stats_entry.max_response_time,
+            'avgContentLength': stats_entry.avg_content_length,
+            'totalRequestsPerSecond': stats_entry.total_rps,
+            'totalFailuresPerSecond': stats_entry.total_fail_per_sec
+        }, **self._get_percentiles_dict(stats_entry)}
+
+    def _get_stats_entries_list(self) -> List[Dict[str, any]]:
+        """Returns a list of dictionaries representing the performance statistics of _all_ Locust tasks that ran
+
+        Returns:
+            List[Dict[str, any]]: A List of Dicts that represent the performance statistics of all Locust tasks
+        """
+        stats = self.locust_env.stats
+        return [self._get_stats_entry_dict(stats_entry) for stats_entry in sort_stats(stats.entries)]
+
+    @property
+    def all_stats(self) -> Dict[str, any]:
+        """A property that returns a snapshot Dict of the current stats of aggregated performance statistics of the current
+        Locust environment.
+
+        Returns:
+            Dict[str, any]: A dictionary of the aggregated performance statistics of all endpoints
+        """
+        return {**{
+            'timestamp': int(time.time()),
+            'tag': self.stats_tag,
+            'environment': self.running_env.name,
+            'statsResetAfterSpawn': self.locust_env.reset_stats,
+            'numUsers': self.locust_env.parsed_options.num_users,
+            'usersPerSecond': self.locust_env.parsed_options.spawn_rate,
+            # We cannot get the user provided runtime directly; however, we can compute a more exact
+            # runtime by subtracting the start time from the last request's time
```

Comment:
Also worth noting: if the test terminates early, this method will be more accurate than the user-provided runtime.